frida/src/script.rs

```diff
@@ -5,45 +5,136 @@
  */
 
 use frida_sys::{FridaScriptOptions, _FridaScript};
+use serde::Deserialize;
+use serde_json::Value;
+use std::cell::Cell;
 use std::marker::PhantomData;
+use std::sync::mpsc::{channel, Receiver, Sender};
 use std::{
     ffi::{c_char, c_void, CStr, CString},
     ptr::null_mut,
 };
 
 use crate::{Error, Result};
 
-unsafe extern "C" fn call_on_message<I: ScriptHandler>(
+const FRIDA_RPC: &str = "frida:rpc";
+
+/// Represents a Frida message.
+#[derive(Deserialize, Debug)]
+#[serde(tag = "type")]
+pub enum Message {
+    /// Represents a message with type send
+    #[serde(alias = "send")]
+    Send(SendMessage),
+    /// Represents a message with type log
+    #[serde(alias = "log")]
+    Log(LogMessage),
+    /// Represents a message with type error
+    #[serde(alias = "error")]
+    Error(ErrorMessage),
+    /// Represents a message with other types
+    #[serde(untagged)]
+    Other(Value),
+}
+
+/// Represents a log message.
+#[derive(Deserialize, Debug)]
+pub struct LogMessage {
+    _level: String,
+    _payload: String,
+}
+
+/// Represents an error message.
+#[derive(Deserialize, Debug)]
+pub struct ErrorMessage {
+    _column_number: usize,
+    _description: String,
+}
+
+/// Represents a message with type send
+#[derive(Deserialize, Debug)]
+pub struct SendMessage {
+    payload: Payload,
+}
+
+/// Represents a message payload.
+#[derive(Deserialize, Debug, Clone)]
+pub struct Payload {
+    /// Message type.
+    pub ty: String,
```

Comment:
maybe make this `r#type`? `ty` is not very rusty.

Reply:
Changed it to r#type :+1: